metadata-ingestion/tests/unit/snowflake/test_snowflake_source.py

```diff
@@ -689,3 +699,111 @@ def test_snowflake_query_result_parsing():
         ],
     }
     assert UpstreamLineageEdge.parse_obj(db_row)
+
+
+class TestDDLProcessing:
+    @pytest.fixture
+    def session_id(self):
+        return "14774700483022321"
+
+    @pytest.fixture
+    def timestamp(self):
+        return datetime.datetime(
+            year=2025, month=2, day=3, hour=15, minute=1, second=43
+        ).astimezone(datetime.timezone.utc)
+
+    @pytest.fixture
+    def extractor(self) -> SnowflakeQueriesExtractor:
+        connection = MagicMock()
+        config = SnowflakeQueriesExtractorConfig()
+        structured_report = MagicMock()
+        filters = MagicMock()
+        structured_report.num_ddl_queries_dropped = 0
+        identifier_config = SnowflakeIdentifierConfig()
+        identifiers = SnowflakeIdentifierBuilder(identifier_config, structured_report)
+        return SnowflakeQueriesExtractor(
+            connection, config, structured_report, filters, identifiers
+        )
+
+    def test_ddl_processing_alter_table_rename(self, extractor, session_id, timestamp):
+        query = "ALTER TABLE person_info_loading RENAME TO person_info_final;"
+        object_modified_by_ddl = {
+            "objectDomain": "Table",
+            "objectId": 1789034,
+            "objectName": "DUMMY_DB.PUBLIC.PERSON_INFO_LOADING",
+            "operationType": "ALTER",
+            "properties": {
+                "objectName": {"value": "DUMMY_DB.PUBLIC.PERSON_INFO_FINAL"}
+            },
+        }
+        query_type = "RENAME_TABLE"
+
+        ddl = extractor.parse_ddl_query(
+            query, session_id, timestamp, object_modified_by_ddl, query_type
+        )
+
+        assert ddl == TableRename(
+            original_urn="urn:li:dataset:(urn:li:dataPlatform:snowflake,dummy_db.public.person_info_loading,PROD)",
+            new_urn="urn:li:dataset:(urn:li:dataPlatform:snowflake,dummy_db.public.person_info_final,PROD)",
+            query=query,
+            session_id=session_id,
+            timestamp=timestamp,
+        ), "Processing ALTER ... RENAME should result in a proper TableRename object"
```

Comment:
in general, we don't need comments on asserts - the test feels pretty clear already